cocotb/log.py

```diff
@@ -86,12 +89,24 @@ def default_config():
 
     # apply level settings for cocotb
     log = logging.getLogger('cocotb')
-    level = os.getenv("COCOTB_LOG_LEVEL", "INFO")
-    try:
+
+    # All log levels are upper case, convert the user input for convenience.
+    level = os.getenv("COCOTB_LOG_LEVEL", logging.getLevelName(_DEFAULT_LOG_LEVEL)).upper()
+
+    # Python does not provide a supported and public function to get the
+    # numeric log level from a string: logging.getLevelName() could be used,
+    # but is apparently unloved for this task. logging._checkLevel() is
+    # private. This leaves us with getattr() as used below, but we need to
+    # prevent getting any property other than the logging constants.
```

Comment:
Can't we just use `log.setLevel(string_from_user)`, and let that handle the error?

Reply:
Namely, I think all we need is 
```python
try:
    level = os.environ['COCOTB_LOG_LEVEL'].upper()
except KeyError:
    level = logging.INFO
log.setLevel(level)  # accepts strings as well as constants
```